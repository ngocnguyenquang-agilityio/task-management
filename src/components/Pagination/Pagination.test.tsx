import { fireEvent, render, screen } from '@testing-library/react';

// Components
import { Pagination } from './index';

describe('Pagination component', () => {
  test('renders the Pagination with children', () => {
    const { container } = render(
      <Pagination
        currentPage={1}
        onChangePageNumber={() => {}}
        pageSize={3}
        siblingCount={2}
        total={50}
        variant="outline"
      />,
    );
    expect(container).toMatchSnapshot();
  });

  test('handles click events', () => {
    const handleClick = jest.fn();
    render(
      <Pagination
        currentPage={1}
        onChangePageNumber={handleClick}
        pageSize={3}
        siblingCount={2}
        total={50}
        variant="outline"
      />,
    );
    fireEvent.click(screen.getByText('1'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  test('disables click events', () => {
    const handleClick = jest.fn();
    render(
      <Pagination
        currentPage={1}
        customClass={{}}
        onChangePageNumber={handleClick}
        pageSize={3}
        siblingCount={2}
        total={50}
        startIcon={<span>Start</span>}
        variant="outline"
      />,
    );
    fireEvent.click(screen.getByText('Previous'));
    expect(handleClick).not.toHaveBeenCalled();
  });
});
