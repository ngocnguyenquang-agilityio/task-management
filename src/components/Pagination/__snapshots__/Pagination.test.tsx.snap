// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component renders the Pagination with children 1`] = `
<div>
  <div
    class="flex flex-row items-center gap-3"
  >
    <button
      class="items-center p-2 disabled:cursor-not-allowed border border-gray-200 hover:bg-gray-100 dark:border-white dark:hover:bg-zinc-100 dark:text-white dark:hover:text-black text-black rounded-lg flex justify-center px-3 bg-zinc-300"
      disabled=""
      type="button"
    >
      <div
        class=""
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
          />
        </svg>
      </div>
      Previous
    </button>
    <button
      class="items-center p-2 disabled:cursor-not-allowed border border-gray-200 hover:bg-gray-100 dark:border-white dark:hover:bg-zinc-100 dark:text-white dark:hover:text-black text-black rounded-lg justify-center w-10 h-10 hidden md:block bg-zinc-300"
      type="button"
    >
      1
    </button>
    <button
      class="items-center p-2 disabled:cursor-not-allowed border border-gray-200 hover:bg-gray-100 bg-transparent dark:border-white dark:hover:bg-zinc-100 dark:text-white dark:hover:text-black text-black rounded-lg justify-center w-10 h-10 hidden md:block"
      type="button"
    >
      2
    </button>
    <button
      class="items-center p-2 disabled:cursor-not-allowed border border-gray-200 hover:bg-gray-100 bg-transparent dark:border-white dark:hover:bg-zinc-100 dark:text-white dark:hover:text-black text-black rounded-lg justify-center w-10 h-10 hidden md:block"
      type="button"
    >
      3
    </button>
    <span
      class="hidden md:block"
    >
      â€¦
    </span>
    <button
      class="items-center p-2 disabled:cursor-not-allowed border border-gray-200 hover:bg-gray-100 bg-transparent dark:border-white dark:hover:bg-zinc-100 dark:text-white dark:hover:text-black text-black rounded-lg justify-center w-10 h-10 hidden md:block"
      type="button"
    >
      17
    </button>
    <button
      class="items-center p-2 disabled:cursor-not-allowed border border-gray-200 hover:bg-gray-100 bg-transparent dark:border-white dark:hover:bg-zinc-100 dark:text-white dark:hover:text-black text-black rounded-lg flex justify-center px-3"
      type="button"
    >
      Next
      <div
        class=""
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M438.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L338.8 224 32 224c-17.7 0-32 14.3-32 32s14.3 32 32 32l306.7 0L233.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160z"
          />
        </svg>
      </div>
    </button>
  </div>
</div>
`;
